src/julia.h

```diff
@@ -275,12 +284,17 @@ typedef struct _jl_datatype_t {
     int32_t ninitialized;
     // hidden fields:
     uint32_t nfields;
-    uint32_t alignment : 31;  // strictest alignment over all fields
+    uint32_t alignment : 29;  // strictest alignment over all fields
     uint32_t haspadding : 1;  // has internal undefined bytes
+    uint32_t fielddesc_type : 2; // 0 -> 8, 1 -> 16, 2 -> 32
     uint32_t uid;
     void *struct_decl;  //llvm::Value*
     void *ditype; // llvm::MDNode* to be used as llvm::DIType(ditype)
-    jl_fielddesc_t fields[];
+    union {
```

Comment:
zero-sized arrays are a GCC-extension anyways (https://msdn.microsoft.com/en-us/library/79wf64bc.aspx). ISO C99 requires that you don't specify the size (https://gcc.gnu.org/onlinedocs/gcc/Zero-Length.html)